### ISSUE TITLE
XLM pre-training using fairseq

### ISSUE THREAD

**Issue by tonylekhtman** [pytorch/fairseq#2065]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
I can't train a XLM model using fairseq.
I wanted to train the xlm_base on 2 languages.
I wanted it to overfit to a small dataset I've created, only 10000 snts in each language.
I've trained it for about 20 epochs but the loss and perplexity were still very high(about 1740 perplexity)
Can you help me find out what's wrong?

#### Code

**Preprocessing:**
The sets are bped using fastBPE, the vocab_mlm is a combination of vocabs in both languages.

DATA_DIR=monolingual_data/fairseq_processed
mkdir -p "$DATA_DIR"

for lg in en zh
do

  fairseq-preprocess \
  --task cross_lingual_lm \
  --srcdict monolingual_data/processed/vocab_mlm \
  --only-source \
  --trainpref monolingual_data/processed/train \
  --validpref monolingual_data/processed/valid \
  --testpref monolingual_data/processed/test \
  --destdir monolingual_data/fairseq_processed \
  --workers 20 \
  --source-lang $lg

  for stage in train test valid
   do
    sudo mv "$DATA_DIR/$stage.$lg-None.$lg.bin" "$DATA_DIR/$stage.$lg.bin"
    sudo mv "$DATA_DIR/$stage.$lg-None.$lg.idx" "$DATA_DIR/$stage.$lg.idx"

  done

done

**Training:**
fairseq-train \
--task cross_lingual_lm monolingual_data/fairseq_processed \
--save-dir checkpoints/mlm_not_legacy \
--max-update 2400000 --save-interval 1 --no-epoch-checkpoints \
--arch xlm_base \
--optimizer adam --lr-scheduler reduce_lr_on_plateau \
--lr-shrink 0.5 --lr 0.0001 --min-lr 1e-09 \
--dropout 0.1 \
--criterion legacy_masked_lm_loss \
--max-tokens 8192 --tokens-per-sample 256 --attention-dropout 0.1 \
--seed 0 \
--monolingual-langs zh,en' --num-segment 2 \
--ddp-backend=no_c10d \
--masked-lm-only

#### What have you tried?
I've tried with 1M,100K,10K snts on each language and got the same results.
#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): 0.9
 - PyTorch Version (e.g., 1.0): 1.4
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): pip install -e .
 - Python version: 3.7
 - CUDA/cuDNN version: 10.2

**Comment by lematt1991** [comment]
Can you provide a gist containing the training log for one of you runs.

**Comment by tonylekhtman** [comment]
<script src="https://gist.github.com/tonylekhtman/ef88ff4c49812c195c619c7e0562687d.js"></script>

**Comment by wannabeOG** [comment]
Hi @tonylekhtman, were you able to fix this particular issue?
